### PR TITLE
Enable extended DNS e2e tests

### PR DESCRIPTION
**Pull request by pravisankar**

### CODE REVIEW

**Comment by pravisankar** [comment]
/hold 
Depends on https://github.com/openshift/origin/pull/21406 (not yet merged)

**Comment by pravisankar** [comment]
/hold cancel 
PR  openshift/origin#21406  got merged

File: ci-operator/config/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ tests:
   container:
     from: src
 - as: e2e-aws
-  commands: TEST_FOCUS='should provide DNS for the cluster' run-tests
+  commands: TEST_FOCUS='DNS' TEST_SKIP='(\[Disabled:.+\]|\[Disruptive\]|\[Skipped\]|\[Slow\]|\[Flaky\]|\[local\]|\[Local\])'
```
Comment (ironcladlou) [inline ci-operator/config/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master.yaml]: @smarterclayton Should these skips be necessary?

Comment (pravisankar) [reply ci-operator/config/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master.yaml]: We need at least `[Disabled:.+]` option to skip the wildcard dns test.

Comment (ironcladlou) [reply ci-operator/config/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master.yaml]: I thought that problematic tests are skipped internally in the extended test setup code.

Comment (ironcladlou) [reply ci-operator/config/openshift/cluster-dns-operator/openshift-cluster-dns-operator-master.yaml]: Maybe I see the problem now.  Only tests who don't match the exclusion list get the suite conformance tags added to them: https://github.com/openshift/origin/blob/master/test/extended/util/test.go#L117

But I think if you're using TEST_FOCUS the internal exclusion list doesn't matter. I still feel like there must be some other way to specify a focus but also use the exclusion list.

Unless @smarterclayton or @stevekuznetsov knows what to do here I say we merge it and revisit.

**Comment by ironcladlou** [comment]
Let's merge this and refactor when we figure out how to avoid the skip stuff.

/lgtm

**Comment by ironcladlou** [comment]
Not entirely sure why this isn't auto-approved based on OWNERS... @smarterclayton @stevekuznetsov ?

**Comment by petr-muller** [comment]
> Not entirely sure why this isn't auto-approved based on OWNERS... @smarterclayton @stevekuznetsov ?

Seems because all affected files do not share an approver who might approve them all, so root `OWNERS` is effective:

```
cat $( find ci-operator/ -name OWNERS | grep -e 'cluster-dns-operator' -e '/coredns/' )
(...pruned comments and reviewers for clarity...)
approvers:
- chrisohaver
- johnbelamaric
- miekg
- yongtang

approvers:
- ironcladlou
- knobunc

approvers:
- chrisohaver
- johnbelamaric
- miekg
- yongtang

approvers:
- ironcladlou
- knobunc
```

**Comment by pravisankar** [comment]
Replaced by smoke suite.

**Pull request closed without merging**